pulser/register.py

```diff
@@ -282,6 +282,29 @@ def _draw_checks(
                     "Needs more than one atom to draw " "the blockade radius."
                 )
 
+    @abstractmethod
+    def _to_dict(self) -> dict[str, Any]:
+        qs = dict(zip(self._ids, map(np.ndarray.tolist, self._coords)))
+        return obj_to_dict(self, qs)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, BaseRegister):
```

Comment:
Isn't this unnecessary considering you then check for `type(self) is type(other)` ?